### PR TITLE
[ssr] Add option to defer hydration on top level custom element

### LINKED ISSUE #2938: [ssr] Add option to defer hydration for top level custom elements

### PR DESCRIPTION
**Pull request by augustjk**
Resolve #2938

### CODE REVIEW

File: packages/labs/ssr/src/lib/render-lit-html.ts [hunk 1]
```diff
@@ -520,6 +520,11 @@ export type RenderInfo = {
    * template, even in the case of conditional templates.
    */
   customElementRendered?: (tagName: string) => void;
+
+  /**
+   * An optional flag to defer hydration of top level custom element
```
Comment (justinfagnani) [inline packages/labs/ssr/src/lib/render-lit-html.ts:525]: Say what the default is?

**Review by justinfagnani — APPROVED** [review]
🎉

**Review by aomarks — APPROVED** [review]

**Review by kevinpschaaf — APPROVED** [review]
It might also be nice to add an integration test for this that ensures `defer-hydration` is actually respected on the top-level element; today we only really test `defer-hydration` at all transitively in the `LitElement` tests at the bottom of `basic.ts`, but not directly.

Testing it in the integration tests would require adding something like an optional `serverRenderOptions` field to `SSRTestDescription` and passing it into [here](https://github.com/lit/lit/blob/main/packages/labs/ssr/src/test/integration/server/server.ts#L52). Then you could add a test to `basic.ts` with `deferHydration: true` that registers an element and `check()`s that e.g. an event handler doesn't actually work yet, and then remove the attribute and see that it does?

That could be in a follow-up PR if we don't want to block this, either way.

**Review by kevinpschaaf — APPROVED** [review]
Looks great!

**Pull request merged**